src/main/java/org/octri/hpoonfhir/controller/MainController.java

```diff
@@ -27,53 +32,83 @@
 
 	@Autowired
 	AnnotationService annotationService;
-	
+
 	@Autowired
 	@Qualifier("r3FhirService")
 	FhirService fhirService;
 
+	@Autowired
+	ObservationAnalysisService observationAnalysisService;
+
 	/**
 	 * Return to a clean search form with no results.
+	 * 
 	 * @param model
 	 * @param request
 	 * @return
 	 */
 	@GetMapping("/")
 	public String home(Map<String, Object> model, HttpServletRequest request) {
+
 		model.put("patientSearchForm", new PatientModel());
 		model.put("results", false);
 		return "search";
 	}
-	
+
 	/**
 	 * Search using the form parameters, and return results.
+	 * 
 	 * @param model
 	 * @param form
 	 * @return
 	 */
-	@PostMapping("/search")
+	@PostMapping("/")
 	public String search(Map<String, Object> model, @ModelAttribute PatientModel form) {
 		model.put("patientSearchForm", form);
 		try {
 			List<Patient> patients = fhirService.findPatientsByFullName(form.getFirstName(), form.getLastName());
-			List<PatientModel> patientModels = patients.stream().map(fhirPatient -> 
-				new PatientModel(fhirPatient.getIdElement().getIdPart(),
-						fhirPatient.getNameFirstRep().getGivenAsSingleString(), 
-						fhirPatient.getNameFirstRep().getFamily())
-			).collect(Collectors.toList());
+			List<PatientModel> patientModels = patients.stream()
+					.map(fhirPatient -> new PatientModel(fhirPatient.getIdElement().getIdPart(),
+							fhirPatient.getNameFirstRep().getGivenAsSingleString(),
+							fhirPatient.getNameFirstRep().getFamily()))
+					.collect(Collectors.toList());
 			model.put("patients", patientModels);
 			model.put("results", true);
 		} catch (FHIRException e) {
-			//TODO: Decide how to handle FHIRException which is only thrown if there's an error in the conversion of V2 messages
+			// TODO: Decide how to handle FHIRException which is only thrown if there's an error in the conversion of V2
+			// messages
 			e.printStackTrace();
 		}
 		return "search";
 	}
 
-
-	@RequestMapping("/labs")
+	/**
+	 * Get the patient observations and convert them to HPO if possible.
+	 * @param model
+	 * @param request
+	 * @return the patient and all associated ObservationModels
+	 */
+	@GetMapping("/labs")
 	public String labs(Map<String, Object> model, HttpServletRequest request) {
-		model.put("patient_id", request.getParameter("patient_id"));
+		String patientId = request.getParameter("patient_id");
+		try {
+			// Get the patient again so information can be displayed
+			Patient fhirPatient = fhirService.findPatientById(patientId);
+			PatientModel patientModel = new PatientModel(fhirPatient.getIdElement().getIdPart(),
+					fhirPatient.getNameFirstRep().getGivenAsSingleString(),
+					fhirPatient.getNameFirstRep().getFamily());
```

Comment:
There is some repetition here with the search method in creating the PatientModel. You might want to consider creating another constructor that takes a single parameter for the fhirPatient.